### PR TITLE
Support negative indices of gather op

### PR DESCRIPTION
**Pull request by BruceDai**
@huningxin @fdwr PTAL, thanks.

### CODE REVIEW

File: src/gather.js [hunk 1]
```diff
@@ -50,8 +50,10 @@ export function gather(input, indices, {axis = 0} = {}) {
     // output[i, j, k, ...] = input[i, j, indices[i, j, k, ...], ...] // if axis == 2
     const outputLoc = output.locationFromIndex(outputIndex);
     const indicesLoc = outputLoc.slice(axis, axis + indices.rank);
-    const selectedInputLoc = outputLoc.slice(0, axis)
-        .concat(indices.getValueByLocation(indicesLoc), outputLoc.slice(axis + indices.rank));
+    let indiceValue = indices.getValueByLocation(indicesLoc);
+    indiceValue = indiceValue < 0 ? indiceValue + input.shape[axis] : indiceValue;
+    const selectedInputLoc =
```
Comment (fdwr) [inline src/gather.js:55]:
```suggestion
    const selectedInputLocation =
```
Whole words please.

**Review by fdwr — APPROVED** [review]
Thanks Bruce.

File: src/gather.js (hunk 1 above)
Comment (BruceDai) [reply src/gather.js]: Done. Please review the new commit, thanks!

**Review by huningxin — APPROVED** [review]
LGTM!

**Review by fdwr — COMMENTED** [review]
👍

**Pull request merged**